src/RequestHeaders.cs

```diff
@@ -18,7 +19,38 @@ public class RequestHeaders : IDictionary<string,IEnumerable<string>> {
     /// </summary>
     /// <param name="headerName">The name of the header to add values to.</param>
     /// <param name="headerValues">The values to add to the header.</param>
-    public void Add(string headerName, params string[] headerValues) {
+    public void Add(string headerName, params string[] headerValues)
+    {
+        if(string.IsNullOrEmpty(headerName))
+            throw new ArgumentNullException(nameof(headerName));
+        if(headerValues == null)
+            throw new ArgumentNullException(nameof(headerValues));
+        if(!headerValues.Any())
+            return;
+        if(_singleValueHeaders.Contains(headerName))
+            _headers[headerName] = new HashSet<string> { headerValues[0] };
+        else if(_headers.TryGetValue(headerName, out var values))
+            foreach(var headerValue in headerValues)
+                values.Add(headerValue);
+        else
+            _headers.Add(headerName, new HashSet<string>(headerValues));
+    }
+    /// <summary>
+    /// Adds values to the header with the specified name if it's not already present
+    /// </summary>
+    /// <param name="headerName">The name of the header to add values to.</param>
+    /// <param name="headerValue">The values to add to the header.</param>
+    public void TryAdd(string headerName, string headerValue)
```

Comment:
We technically already have a try add method for dictionaries, so maybe we could rely on this implementation here? https://github.com/microsoft/kiota-abstractions-dotnet/blob/main/src/extensions/IDictionaryExtensions.cs#L25

Also I'm not sure how the resolution between that extension method and a local method is going to work. Maybe @andrueastman can share insights here

Reply:
The second parameter will be different. The disctionary here has type `<string, IEnumerable<string>` so the extension will expect `IEnumerable<string>` when we are taking in a `string` here. So this will be technically just be an overload.